src/lib.rs

```diff
@@ -99,6 +101,46 @@ impl Environment {
         self
     }
 
+    /// An iterator visiting all key-value pairs. The iterator element type is (key, value).
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate environment;
+    ///
+    /// use std::ffi::OsString;
+    ///
+    /// let env = environment::Environment::empty().insert("FOO", "BAR");
+    ///
+    /// let mut iterator = env.iter();
+    ///
+    /// assert_eq!(iterator.next(), Some((&"FOO".into(), &"BAR".into())));
+    /// ```
+    pub fn iter(&self) -> Iter<OsString, OsString> {
+        self.vars.iter()
+    }
+
+    /// Returns an iterator that allows modifying each value.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate environment;
+    ///
+    /// use std::ffi::OsString;
+    ///
+    /// let mut env = environment::Environment::empty().insert("FOO", "BAR");
+    ///
+    /// for (_, v) in env.iter_mut() {
+    ///     *v = OsString::from("BAZ");
+    /// }
+    ///
+    /// assert_eq!(env, vec![("FOO", "BAZ")].into());
+    /// ```
+    pub fn iter_mut(&mut self) -> IterMut<OsString, OsString> {
```

Comment:
I assumed there wouldn't be an `iter_mut`, that the main way of mutating is through the fluent API.  *shrug*

Reply:
Yep, that's right.